### DISCUSSION TITLE
Should the Cantata cask become a core formula?

### DISCUSSION THREAD

**Discussion by Marcool04** [Homebrew/discussions#1370] [Everyday usage] [answered]
Hi all,

I have just spend a few hours getting the Qt5 Graphical MPD Client Cantata (upstream: https://github.com/cdrummond/cantata) to build manually. This was prompted by the realization that the version I had been running, installed via the homebrew cask for Cantata: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/cantata.rb, has been outdated for over two years: current version installed by the cask is 2.3.2, which was released on August 2nd, 2018, see the upstream release page: https://github.com/CDrummond/cantata/tags. Latest version is 2.4.3. The reason for this is that the developer of Cantata is no longer providing .dmg builds, contrary to what was the case a few years back, see his responses in this: https://github.com/CDrummond/cantata/issues/1671, this: https://github.com/CDrummond/cantata/issues/1543 and this: https://github.com/CDrummond/cantata/issues/1534 issue.

There have been new distributions since then, but no mac nor windows builds.

I noticed the discussion here: https://github.com/Homebrew/homebrew-core/pull/3331 regarding gedit, which used to be a cask, and was migrated to being a formula specifically because upstream stopped providing builds for mac. This is explicitly mentioned as an example of something that is not an acceptable cask and should be a formula, on the "Acceptable casks" page: https://github.com/Homebrew/brew/blob/master/docs/Acceptable-Casks.md#rejected-casks

> The app is open-source [yes] and has a GUI [yes] but no compiled versions (or only old ones [indeed]) are provided. **It’s better to have them in Homebrew so users don’t get perpetually outdated versions**. See [gedit](https://github.com/Homebrew/homebrew-cask/pull/23360) for example.

So, here is my question: is it time to make a formula for Cantata? If so, I'd be happy to contribute what I learned from building it manually. It's tightly bound to qt5 which, I discovered, is a bit of a pain, but it can be done. This is a relatively well used app, even in the form of the outdated cask:

````
$ brew info Cantata
cantata: 2.3.2
https://github.com/cdrummond/cantata
/usr/local/Caskroom/cantata/2.3.2 (64B)
From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/cantata.rb
==> Name
Cantata
==> Description
Qt5 Graphical MPD Client
==> Artifacts
Cantata.app (App)
==> Analytics
install: 17 (30 days), 67 (90 days), 243 (365 days)
````

Thanks for any input.
Regards,
Mark.

**Comment by carlocab** [comment, marked as answer]
Feel free to open a PR to add Cantata to homebrew/core. Reading these might help:

https://docs.brew.sh/Formula-Cookbook
https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md

**Comment by Marcool04** [comment]
Thanks for the quick response :)
I was just worried about this part of the "Acceptable Formulae" page: https://docs.brew.sh/Acceptable-Formulae

> Stuff that builds an .app
> Don’t make your formula build an .app (native macOS Application); we don’t want those things in Homebrew. Encourage upstream projects to build and support a .app that can be distributed by homebrew/cask (and used without it, too).

Would this qualify as an exception?

**Reply by carlocab** [reply]
Possibly. Only way to find out is to open a PR. 🙂

Even if it's not accepted, that will help you refine a formula to host in an external tap, which will allow you to build bottles for it too.

**Comment by Marcool04** [comment]
Right. Good point. Will get to it then 😄

**Comment by aeleeds** [comment]
@Marcool04 did you make a formula for Cantata? Or do you still intend to? I would love to be using an up to date version.

**Reply by Marcool04** [reply]
Hi @aeleeds, I must admit: I tryed, then gave up. I knew nothing about ruby when I started, learned a great deal, but couldn't get it to build properly…
I still build Cantata manually from the head of the github repo, which works, usually with some qt5 / qt6 mess-ups.
Sorry to disappoint.
Regards,
Mark.
